framework/src/main/java/org/checkerframework/framework/ajava/InsertAjavaAnnotations.java

```diff
@@ -0,0 +1,580 @@
+                existingImports.add(printer.print(importDecl));
+            }
+
+            List<String> newImports = new ArrayList<>();
+            for (ImportDeclaration importDecl : node1.getImports()) {
+                String importString = printer.print(importDecl);
+                if (!existingImports.contains(importString)) {
+                    newImports.add(importString);
+                }
+            }
+
+            if (!newImports.isEmpty()) {
+                int position;
+                int lineBreaks;
+                if (!node2.getImports().isEmpty()) {
+                    Position lastImportPosition =
+                            node2.getImports().get(node2.getImports().size() - 1).getEnd().get();
+                    position = getAbsolutePosition(lastImportPosition) + 1;
+                    lineBreaks = 1;
+                } else if (node2.getPackageDeclaration().isPresent()) {
+                    Position packagePosition = node2.getPackageDeclaration().get().getEnd().get();
+                    position = getAbsolutePosition(packagePosition) + 1;
+                    lineBreaks = 2;
+                } else {
+                    position = 0;
+                    lineBreaks = 0;
+                }
+
+                String insertionContent = String.join("", newImports);
+                for (int i = 0; i < lineBreaks; i++) {
+                    insertionContent = System.lineSeparator() + insertionContent;
+                }
+
+                insertions.add(new Insertion(position, insertionContent));
+            }
+
+            node1.getModule().ifPresent(l -> l.accept(this, node2.getModule().get()));
+            node1.getPackageDeclaration()
+                    .ifPresent(l -> l.accept(this, node2.getPackageDeclaration().get()));
+            for (int i = 0; i < node1.getTypes().size(); i++) {
+                node1.getTypes().get(i).accept(this, node2.getTypes().get(i));
+            }
+        }
+
+        /**
+         * Creates an insertion for a collection of annotations. The annotations will appear on
+         * their own line unless any non-whitespace characters precede the insertion position on its
+         * own line.
+         *
+         * @param position the position of the insertion
+         * @param annotations List of annotations to insert
+         */
+        private void addAnnotationOnOwnLine(Position position, List<AnnotationExpr> annotations) {
+            String line = lines.get(position.line - 1);
+            int insertionColumn = position.column - 1;
+            boolean ownLine = true;
+            for (int i = 0; i < insertionColumn; i++) {
+                if (line.charAt(i) != ' ' && line.charAt(i) != '\t') {
+                    ownLine = false;
+                    break;
+                }
+            }
+
+            if (ownLine) {
+                StringBuilder insertionContent = new StringBuilder();
```

Comment:
Use StringJoiner instead of StringBuilder, and the body of the `for` loop can be 1 line.